src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -75,19 +82,27 @@ protected override int PublishMessages(List<QueueWithItems<KafkaItem>> itemsPerT
 
                 } while (true);
             }
-            catch (Exception e)
+            catch (KafkaException e)
             {
                 producer.Dispose();
+                var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Failed to initialize transactions for the producer instance. Error code: '{e.Error.Code}'. Error reason: '{e.Error.Reason}'{Environment.NewLine}");
+                if (e.Error.Code == ErrorCode.TransactionalIdAuthorizationFailed)
+                {
+                    errorMessageBuilder.Append(
+                        $"Add an required ACL permissions to your API Key that will make WRITE and DESCRIBE operations possible for transactional ID '{TransactionalId}'. ");
+                }
+                else
+                {
+                    errorMessageBuilder.Append($"If you are using a single node Kafka cluster then the following settings might be required:{Environment.NewLine}" +
+                                               $"- transaction.state.log.replication.factor: 1 {Environment.NewLine}" +
+                                               "- transaction.state.log.min.isr: 1");
+                }
 
-                string msg = $" ETL process: {Name}. Failed to initialize transactions for the producer instance. " +
-                             $"If you are using a single node Kafka cluster then the following settings might be required:{Environment.NewLine}" +
-                             $"- transaction.state.log.replication.factor: 1 {Environment.NewLine}" +
-                             "- transaction.state.log.min.isr: 1";
-
-                if (Logger.IsOperationsEnabled) 
-                    Logger.Operations(msg, e);
-
-                throw new QueueLoadException(msg, e);
+                throw new QueueLoadException(errorMessageBuilder.ToString(), e);
+            }
+            catch (Exception e)
+            {
```

Comment:
Missing disposal of `producer`

Reply:
done https://github.com/ravendb/ravendb/pull/17577/commits/640d4f4056e6e9d834f0db6d03b65f619d19c2ad